Integrations/FeedTAXII/FeedTAXII.py

```diff
@@ -511,6 +517,31 @@ def __init__(self, insecure: bool = True, polling_timeout: int = 20, initial_int
                 self.username = credentials.get('identifier', None)
                 self.password = credentials.get('password', None)
 
+        self.all_collections = self.get_all_collections()
+        if collection is None or collection == '':
+            return_error(f"No collection set. Here is a list of all accessible collections: "
+                         f"{str(self.all_collections)}")
```

Comment:
It seems like this is the only place you're using `self.all_collections`. If you're not using it I think it doesn't need to be a member. While at it, please move the call to `get_all_collections()` inside the condition scope, since we can save the request time it if the user provided a collection.

Reply:
Well, I kept it that way so I can validate it in the test module as well.